Doc/library/itertools.rst

```diff
@@ -464,6 +465,25 @@ loops that truncate the stream.
                       pass
 
 
+.. function:: ilen(iterable)
+
+   Return the length of an *iterable* by counting iterations.
```

Comment:
I think it is obvious that you will count iterations to count the number of elements of an iterable (though, if you say "iterations" people might wonder "iterations of what"). So "Return the length of an *iterable*" should be sufficient.

Reply:
What about "Iterate over *iterable* and return iteration count."?

Reply:
I still think it's better to just explain what it does in simpler terms rather in a convoluted way here. "Return the length of the iterable" is straightforward and we explain just after that an iterator is consumed.

Reply:
But if the input is a `list`, then the user might be confused why it is taking so long, this will give a clearer signal that it achieves it by iterating over it. This would address your previous concern better, to notify the user that it is not as efficient as `len` for `Sequences`.

Reply:
Then you should mention this in a separate line, not in the description of the function itself.